build_runner/test/integration_tests/utils/build_descriptor.dart

```diff
@@ -161,7 +161,7 @@ ${builders.map(_builderFactory).join('\n')}
 ''';
 
 String _builderFactory(TestBuilderDefinition builder) =>
-    'Builder ${builder.key}Factory(_) => ${builder.key};';
+    'Builder ${builder.key}Factory(BuilderOptions _) => ${builder.key};';
```

Comment:
I don't think there is any value in this case?

Reply:
Yeah it's really low priority, I can drop the PR. I only changed it because it showed up in a regex search and if I were writing this line today I'd do it this way to match our other code.

Reply:
I was mainly looking for this pattern in user facing docs because those would be worth fixing, but it doesn't look like we have any.

Reply:
hmm ok, I sort of prefer it the other way even though it ends up as dynamic... the whole point of the `_`  pattern is to reduce boilerplate and the type isn't useful information here imo.

I am definitely on board with changing it if we do want to add types, I just don't think I agree that's a pattern I like?

Reply:
It would be sort of interesting if for wildcards the type defaulted to `void` or something like that. I am not sure what it actually defaults to (probably still dynamic?).

Reply:
I'm not sure if I'm 100% sold on the pattern, but it's the style we landed on to apply across the Dart and Flutter teams when we implemented the strict top level inference lint. It would apply here if this were handwritten and analyzed code.

Reply:
> but it's the style we landed on to apply across the Dart and Flutter teams when we implemented the strict top level inference lint

But the lint explicitly doesn't cover local functions like this - it seems to only look at public API. I am curious what the purpose of it is.

Reply:
> But the lint explicitly doesn't cover local functions like this

This is writing a top level function into a generated file.

Reply:
Ohhhhh, I see this is actually a string which is being emitted at the top level, so it is not a local function. Nevermind, LGTM.